openslides/mediafiles/static/js/mediafiles/site.js

```diff
@@ -331,12 +331,27 @@ angular.module('OpenSlidesApp.mediafiles.site', ['ngFileUpload', 'OpenSlidesApp.
                 return Upload.upload({
                     url: '/rest/mediafiles/mediafile/',
                     method: 'POST',
-                    data: {mediafile: mediafile.newFile, title: mediafile.title, uploader_id: mediafile.uploader_id}
+                    data: {mediafile: mediafile.newFile, title: mediafile.title, uploader_id: mediafile.uploader_id, private: mediafile.private}
                 });
 
             }
         };
     }
+])
+
+.filter('privateFilter', [
+    '$filter',
+    'operator',
+    function ($filter, operator) {
+        return function (array) {
+            if (operator.hasPerms('mediafiles.can_see_private')) {
+                return array;
+            }
+            return Array.prototype.filter.call(array, function (item) {
```

Comment:
I think that should actually work in this case. I believe I took this approach from somewhere else in the codebase or online, and is usually what you want to use if you have array-likes rather than actual arrays. So if the files are always an array, that approach is possible.